ports/esp32/machine_timer.c

```diff
@@ -130,20 +136,44 @@ STATIC void machine_timer_enable(machine_timer_obj_t *self) {
 
 STATIC mp_obj_t machine_timer_init_helper(machine_timer_obj_t *self, mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
     static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_period,       MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 0xffffffff} },
         { MP_QSTR_mode,         MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 1} },
         { MP_QSTR_callback,     MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+#if MICROPY_PY_MACHINE_HI_RES_TIMER
+        { MP_QSTR_period,       MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+        { MP_QSTR_freq,         MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+#else
+        { MP_QSTR_period,       MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 0xffffffff} },
+#endif
     };
 
     machine_timer_disable(self);
 
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
+#if MICROPY_PY_MACHINE_HI_RES_TIMER
+    if (args[3].u_obj != mp_const_none) {
+        mp_float_t freq;
+        if (mp_obj_get_float_maybe(args[3].u_obj, &freq)) {
```

Comment:
To minimise code, I'd usually just use `mp_obj_get_float()`.  That will raise an exception if the arg is not a float or int.  Yes it'll be a generic exception ("can't convert <type> to float") but I think that's fair enough, no real need to have a specific exception here.  The only time you'd trigger the exception is if you passed in a list or something silly.

Reply:
Done. For some reason I thought `mp_obj_get_float()` would fail if the input was an `int` but looking at the code this is obviously not the case.